Modules/signalmodule.c

```diff
@@ -3,6 +3,13 @@
 
 /* XXX Signals should be recorded per thread, now we have thread state. */
 
+#ifdef MS_WINDOWS
+#  if !defined(SOCKET) && defined(Py_INTERNAL_SIGNAL_H)
+#    error "pycore_signal.h included without PYCORE_SIGNAL_WITH_PRE_INCLUDES"
+#  endif
+#  define PYCORE_SIGNAL_WITH_PRE_INCLUDES
+#endif
```

Comment:
@zooba, does this match what you were recommending?

Reply:
This should be an impossible situation because there are no preceding includes.

Line 8 is also an impossible situation _after_ including `internal_signal.h` because there will have been an error already.